### PR TITLE
Update Analytics to use `GTag` infrastructure

### PR DESCRIPTION
**Pull request by nfmohit**
## Summary

Addresses issue:

- #8273 

## Relevant technical choices

- This PR updates the Analytics module to use the new `GTag` infrastructure.

## PR Author Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 5.2 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

File: tests/phpunit/integration/Modules/Analytics_4Test.php
Comment (aaemnnosttv) [inline tests/phpunit/integration/Modules/Analytics_4Test.php:2801]: This doesn't need to be called conditionally, does it?

Comment (nfmohit) [reply tests/phpunit/integration/Modules/Analytics_4Test.php]: You're right, it doesn't and is unnecessary. I've removed it, thank you for highlighting!

File: includes/Modules/Analytics_4/Web_Tag.php
Comment (aaemnnosttv) [inline includes/Modules/Analytics_4/Web_Tag.php:127]: Let's use a proper type hint for the parameter as well.
```suggestion
	protected function setup_gtag( GTag $gtag ) {
```

File: includes/Modules/Analytics_4/Web_Tag.php
Comment (aaemnnosttv) [inline includes/Modules/Analytics_4/Web_Tag.php:144]: This has the unexpected effect of changing the order in which the `linker` is set. In my testing, I saw it as the last command, where apparently the order can be significant.

See https://developers.google.com/tag-platform/devguides/cross-domain#set_up_cross-domain_linking

Let's update it to remain as the first command, when set.

File: includes/Core/Tags/GTag.php
Comment (aaemnnosttv) [inline includes/Core/Tags/GTag.php:197]: As a public method, please add test coverage.

For example, this looks like it would error if called with no `tags` added.

**Review by aaemnnosttv — CHANGES_REQUESTED** [review]
Thanks @nfmohit – this looks good, just a few minor things to address.

More importantly though, the AC calls for this change to only happen conditionally.
> * With the `adsModule` feature flag enabled

I'm not sure if this is really necessary, but it should be clarified and resolved before we merge this.

File: includes/Modules/Analytics_4/Web_Tag.php
Comment (nfmohit) [reply includes/Modules/Analytics_4/Web_Tag.php]: Added, thank you for the tip!

File: includes/Modules/Analytics_4/Web_Tag.php
Comment (nfmohit) [reply includes/Modules/Analytics_4/Web_Tag.php:144]: Thank you for pointing this out. I've addressed this by outputting all the commands (added via `add_command`) before the hardcoded `gtag("js")` command. Please let me know if that sounds good, thanks!

File: includes/Core/Tags/GTag.php
Comment (nfmohit) [reply includes/Core/Tags/GTag.php:198]: This got missed, my bad. Added now, thanks!

**Comment by nfmohit** [comment]
> Thanks @nfmohit – this looks good, just a few minor things to address.

I have addressed all the comments, please let me know if all sounds good, thank you!

> More importantly though, the AC calls for this change to only happen conditionally.
> 
> > * With the `adsModule` feature flag enabled
> 
> I'm not sure if this is really necessary, but it should be clarified and resolved before we merge this.

Ah, my apologies, this important detail completely slipped through, I think mostly because this isn't directly related to the Ads module and should be able to function independently. I'll wait for @techanvil's feedback in the issue. Thank you for spotting!

File: includes/Modules/Analytics_4/Web_Tag.php
Comment (aaemnnosttv) [reply includes/Modules/Analytics_4/Web_Tag.php:144]: Thanks, although that change would also print the `config` commands and maybe others before `js`. I think we want to keep the order as today for now, and we can see about building in some kind of priority sorting maybe in the future.

**Review by aaemnnosttv — COMMENTED** [review]
Thanks @nfmohit , one more reply to address and then I think this should be good to go.

File: includes/Modules/Analytics_4/Web_Tag.php
Comment (nfmohit) [reply includes/Modules/Analytics_4/Web_Tag.php:144]: Thank you!

The `config` commands do appear after `js` with the current ordering in place in this PR. See:

image

This imitates the order that we have today. I absolutely agree, some kind of mechanism to sort these by priority and position would be great, I can open an issue to track that.

Do you think what we have in this PR is sufficient for now?

Comment (aaemnnosttv) [reply includes/Modules/Analytics_4/Web_Tag.php:144]: My mistake, thanks for clarifying! 👍

**Review by aaemnnosttv — APPROVED** [review]
LGTM, thanks @nfmohit !

**Pull request merged**
